### PR TITLE
Working Joe updates & Predator Spawning

### PR DESCRIPTION
**Pull request by realforest2001**
# About the pull request
Reorders checks for spawning as a Predator, so it tells you if you even can spawn BEFORE you confirm if you want to or not.

Updates the Working Joe skills to have SKILL_DOMESTIC.
Rebrands Binary language as Apollo Link, used by Legacy robots & drones, but also by Working Joes.
Added a verb for staff to send messages as ARES on the Apollo Link.
Changes the message on WJ spawn to use general comms rather than the intrusive broadcast.

# Explain why it's good for the game
Adds a bit more depth to Working Joes, opening opportunities for events and allowing more coordination. Also opens the door for things using Maintenance Drones.
Makes joining the hunt run checks more effectively.

# Testing Photographs and Procedure

Ensured appropriate people can hear Apollo Link, and that the whitelist flags are appropriate.
Reordered Join-the-Hunt checks work as intended.
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog

:cl:
code: Reordered checks for joining the hunt.
add: Added backend whitelist flag for Working Joe, allowing staff to use WJ without having Synthetic.
add: Rebranded Binary Language as Apollo Link, an ARES subprocessor that WY definitely did not steal from Seegson.
add: Changed the Working Joe late-join broadcast to use general comms rather than an intrusive announcement.
admin: Added a verb for staff to talk as ARES on the Apollo Link.
del: Removed Drone Talk as an unnecessary language. 
/:cl:

### CODE REVIEW

**Comment by ThePiachu** [comment]
Make sure to update https://cm-ss13.com/wiki/Comms when merged

File: code/game/gamemodes/cm_initialize.dm [hunk 1]
```diff
@@ -195,6 +192,9 @@ Additional game mode variables.
 			to_chat(pred_candidate, SPAN_WARNING("You already were a Yautja! Give someone else a chance."))
 		return
 
+	if(show_warning && alert(pred_candidate, "Confirm joining the hunt. You will join as \a [lowertext(J.get_whitelist_status(RoleAuthority.roles_whitelist, pred_candidate.client))] predator", "Confirmation", "Yes", "No") == "No")
```
Comment (stanalbatross) [inline code/game/gamemodes/cm_initialize.dm:195]: needs to be != "Yes"

also maybe switch to tgui alert

File: code/game/objects/items/devices/radio/encryptionkey.dm [hunk 2]
```diff
@@ -14,7 +14,7 @@
 
 /obj/item/device/encryptionkey/binary
 	icon_state = "binary_key"
-	translate_binary = 1
+	translate_apollo = 1
```
Comment (stanalbatross) [inline code/game/objects/items/devices/radio/encryptionkey.dm:17]: bools not ints

File: code/modules/admin/tabs/event_tab.dm [hunk 3]
```diff
@@ -579,6 +579,36 @@
...
+	set category = "Admin.Factions"
+
+	if(!admin_holder || !(admin_holder.rights & R_MOD))
+		to_chat(src, "Only administrators may use this command.")
+		return
+	var/input = input(usr, "This is a broadcast from the ship AI to Working Joes and Maintenance Drones. Do not use html.", "What?", "") as message|null
+	if(!input)
+		return FALSE
+
+	for(var/obj/structure/machinery/computer/almayer_control/C in machines)
```
Comment (stanalbatross) [inline code/modules/admin/tabs/event_tab.dm:593]: varname

File: code/modules/admin/tabs/event_tab.dm [hunk 4]
```diff
@@ -579,6 +579,36 @@
...
+		if(!(C.inoperable()))
```
Comment (stanalbatross) [inline code/modules/admin/tabs/event_tab.dm:594]: early return + tochat here for readability pls

File: code/modules/mob/language/languages.dm [hunk 5]
```diff
@@ -145,48 +145,45 @@
...
-	GLOB.STUI.game.Add("\[[time_stamp()]]<font color='#FFFF00'>BINARY: [key_name(speaker)] : [message]</font><br>")
+	var/message_body = "<span class='message'>broadcasts, \"[message]\"</span></span></i>"
+	GLOB.STUI.game.Add("\[[time_stamp()]]<font color='#FFFF00'>APOLLO: [key_name(speaker)] : [message]</font><br>")
 	GLOB.STUI.processing |= STUI_LOG_GAME_CHAT
 	for (var/mob/M in GLOB.dead_mob_list)
 		if(!istype(M,/mob/new_player) && !istype(M,/mob/living/brain)) //No meta-evesdropping
-			M.show_message("[message_start] [message_body]", SHOW_MESSAGE_AUDIBLE)
+			M.show_message("<span class='[colour]'>[message_start] [message_body]</span>", SHOW_MESSAGE_AUDIBLE)
 
 	for (var/mob/living/S in GLOB.alive_mob_list)
```
Comment (stanalbatross) [inline code/modules/mob/language/languages.dm:173]: varnames...

**Review by stanalbatross — CHANGES_REQUESTED** [review]

File: code/game/jobs/job/civilians/support/working_joe.dm [hunk 6]
```diff
@@ -6,12 +6,12 @@
...
 	selection_class = "job_working_joe"
 	flags_startup_parameters = ROLE_ADD_TO_DEFAULT|ROLE_WHITELISTED|ROLE_CUSTOM_SPAWN
-	flags_whitelist = WHITELIST_SYNTHETIC
+	flags_whitelist = WHITELIST_JOE
 	gear_preset = /datum/equipment_preset/synth/working_joe
 	entry_message_body = "You are a Working Joe!  You are held to a higher standard and are required to obey not only the Server Rules but Marine Law and Synthetic Rules.  Failure to do so may result in your White-list Removal.  Your primary job is to maintain the cleanliness of the ship, putting things in their proper place.  Your capacities are limited, but you have all the equipment you need, and the central AI has a plan!"
 
 /datum/job/civilian/working_joe/announce_entry_message(mob/living/carbon/human/H)
-	addtimer(CALLBACK(GLOBAL_PROC, .proc/all_hands_on_deck, "ARES has deployed [H.real_name] to help with operations."), 1.5 SECONDS)
+	addtimer(CALLBACK(GLOBAL_PROC, .proc/ai_announcement, "[H.real_name] has been deployed to help with operations."), 1.5 SECONDS)
```
Comment (harryob) [inline code/game/jobs/job/civilians/support/working_joe.dm:14]: GLOBAL_PROC_REF macro

**Review by harryob — CHANGES_REQUESTED** [review]

**Review by harryob — APPROVED** [review]

**Review by stanalbatross — APPROVED** [review]

**Pull request merged**